pkg/skaffold/version/version_test.go

```diff
@@ -57,12 +57,69 @@ func TestParseVersion(t *testing.T) {
 }
 
 func TestUserAgent(t *testing.T) {
-	testutil.Run(t, "", func(t *testutil.T) {
-		t.Override(&platform, "osx")
-		t.Override(&version, "1.0")
+	tests := []struct {
+		description string
+		platform    string
+		version     string
+		expected    string
+	}{
+		{
+			description: "version number specified",
+			platform:    "osx",
+			version:     "1.0",
+			expected:    "skaffold/1.0 (osx)",
+		},
+		{
+			description: "version not number specified",
+			platform:    "osx",
+			expected:    "skaffold (osx)",
+		},
```

Comment:
Since we may be using these values in computing metrics, it's safer to add checks for `intellij` and `gcloud` too so that tests will break on accidental changes.